mapviz_plugins/src/point_drawing_plugin.cpp

```diff
@@ -0,0 +1,485 @@
+      glVertex2f(cur_point_.transformed_arrow_point.getX(),
+                 cur_point_.transformed_arrow_point.getY());
+
+      glVertex2f(cur_point_.transformed_arrow_point.getX(),
+                 cur_point_.transformed_arrow_point.getY());
+      glVertex2f(cur_point_.transformed_arrow_left.getX(),
+                 cur_point_.transformed_arrow_left.getY());
+
+      glVertex2f(cur_point_.transformed_arrow_point.getX(),
+                 cur_point_.transformed_arrow_point.getY());
+      glVertex2f(cur_point_.transformed_arrow_right.getX(),
+                 cur_point_.transformed_arrow_right.getY());
+
+      transformed = true;
+    }
+
+    glEnd();
+
+    return transformed;
+  }
+  bool PointDrawingPlugin::TransformPoint(StampedPoint& point)
+  {
+    swri_transform_util::Transform transform;
+    if (GetTransform(point.source_frame, point.stamp, transform))
+    {
+      point.transformed_point = transform * point.point;
+
+      tf::Transform orientation(tf::Transform(transform.GetOrientation()) *
+                                point.orientation);
+      point.transformed_arrow_point =
+          point.transformed_point + orientation * tf::Point(1.0, 0.0, 0.0);
+      point.transformed_arrow_left =
+          point.transformed_point + orientation * tf::Point(0.75, -0.2, 0.0);
+      point.transformed_arrow_right =
+          point.transformed_point + orientation * tf::Point(0.75, 0.2, 0.0);
+
+      if (covariance_checked_)
+      {
+        for (uint32_t i = 0; i < point.cov_points.size(); i++)
+        {
+          point.transformed_cov_points[i] = transform * point.cov_points[i];
+        }
+      }
+
+      point.transformed = true;
+      return true;
+    }
+
+    point.transformed = false;
+    return false;
+  }
+
+  void PointDrawingPlugin::Transform()
+  {
+    bool transformed = false;
+
+    std::list<StampedPoint>::iterator points_it = points_.begin();
+    for (; points_it != points_.end(); ++points_it)
+    {
+      transformed = transformed | TransformPoint(*points_it);
+    }
+
+    transformed = transformed | TransformPoint(cur_point_);
+    if (laps_.size() > 0)
+    {
+      for (int i = 0; i < laps_.size(); i++)
+      {
+        std::list<StampedPoint>::iterator lap_it = laps_[i].begin();
+        for (; lap_it != laps_[i].end(); ++lap_it)
+        {
+          transformed = transformed | TransformPoint(*lap_it);
+        }
+      }
+    }
+    if (!points_.empty() && !transformed)
+    {
+      PrintError("No transform between " + cur_point_.source_frame + " and " +
+                 target_frame_);
+    }
+  }
+
+  bool PointDrawingPlugin::DrawLaps()
+  {
+    bool transformed = false;
+    glColor4f(color_.redF(), color_.greenF(), color_.blueF(), 0.5);
+    glLineWidth(3);
+    QColor base_color = color_;
+    if (laps_.size() != 0)
+    {
+      for (int i = 0; i < laps_.size(); i++)
+      {
+        int hue = color_.hue() + (i + 1) * 10 * 3.1415926;
```

Comment:
Agreed; I just assumed it was being brought in somewhere. If it's not, `M_PI` is better.

> On Jun 24, 2016, at 21:32, Edward Venator <notifications@github.com> wrote:
> 
> In mapviz_plugins/src/point_drawing_plugin.cpp:
> 
> > +      PrintError("No transform between " + cur_point_.source_frame + " and " +
> > +                 target_frame_);
> > +    }
> > +  }
> > +
> > +  bool PointDrawingPlugin::DrawLaps()
> > +  {
> > +    bool transformed = false;
> > +    glColor4f(color_.redF(), color_.greenF(), color_.blueF(), 0.5);
> > +    glLineWidth(3);
> > +    QColor base_color = color_;
> > +    if (laps_.size() != 0)
> > +    {
> > +      for (int i = 0; i < laps_.size(); i++)
> > +      {
> > +        int hue = color_.hue() + (i + 1) * 10 * 3.1415926;
> I'd recommend M_PI over swri_math_util::_pi because swri_math_util is not currently a dependency of mapviz, and it seems unnecessary to bring it in just for a constant.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or mute the thread.
>